src/core/ComposableElement.php

```diff
@@ -466,31 +466,8 @@ protected function init(): void {}
       for ($i = 0; $i < $ln; ++$i) {
         $child = $this->children->get($i);
         if ($child instanceof :x:element) {
-          do {
-            assert($child instanceof :x:element);
-            if ($child instanceof XHPAwaitable) {
-              $child = static::__xhpAsyncRender($child)->getWaitHandle();
-            } else {
-              $child = $child->render();
-            }
-            if ($child instanceof WaitHandle) {
-              $childWaitHandles[$i] = $child;
-            } else if ($child instanceof :x:element) {
-              continue;
-            } else if ($child instanceof :x:frag) {
-              $children = $this->children->toValuesArray();
-              array_splice($children, $i, 1, $child->getChildren());
-              $this->children = new Vector($children);
-              $ln = count($this->children);
-              --$i;
-            } else if ($child === null) {
-              $this->children->removeKey($i);
-              $i--;
-            } else {
-              assert($child instanceof XHPChild);
-              $this->children[$i] = $child;
-            }
-          } while ($child instanceof :x:element);
+          $child = $child->__renderAndProcess()->getWaitHandle();
+          $childWaitHandles[$i] = $child;
```

Comment:
I'm confused. Where did all the logic go? What happens when $child returns an `<x:frag>` and the children need to be appended to the parent's children?

```
$child = $this->children->get($i);
$child = $child->__renderAndProcess()->getWaitHandle();
```
Does that mean that `$this->children->get($i)` is now the rendered child? I thought it didn't.

Also, the way I did it this was was for situations like this:

```
// <awaitable-group> just renders as <x:frag><xhp-awaitable /><xhp-awaitable /></x:frag>
<div>
  <awaitable-xhp />
  <awaitable-group />
</div>
```

If you await each child, then you will go through two await phases, but if you loop through each rendered child, and re-render it if doesn't await, then you can do all three `<awaitable-xhp>` elements in one await.